File: python/examples/manage_app_git_connection.py
import os

from stackmachine import StackMachine

app_id = "da_example"
github_installation_repo_id = "github_repo_example"
deploy_branch = "main"

with StackMachine(os.environ["STACKMACHINE_API_KEY"]) as client:
    current = client.apps.git.retrieve(app_id)
    print("Current Git connection:", current)

    connection = client.apps.git.connect(
        app=app_id,
        installation_repo_id=github_installation_repo_id,
        deploy_branch=deploy_branch,
    )
    print("Connected Git repository:", connection.id)

    updated = client.apps.git.update(
        connection.id,
        deployment_status_events=True,
        pull_request_comments=True,
    )
    print("Updated Git connection:", updated.id)

    # Uncomment to disconnect the repository from the app.
    # client.apps.git.delete(app_id)

File: js/examples/manageAppGitConnection.js
import { StackMachine } from "stackmachine";

const STACKMACHINE_API_KEY = process.env.STACKMACHINE_API_KEY;

if (!STACKMACHINE_API_KEY) {
  throw new Error("Set STACKMACHINE_API_KEY before running this example.");
}

const client = new StackMachine(STACKMACHINE_API_KEY);

const appId = "da_example";
const githubInstallationRepoId = "github_repo_example";
const deployBranch = "main";

const current = await client.apps.git.retrieve(appId);
console.log("Current Git connection:", current);

const connection = await client.apps.git.connect({
  app: appId,
  installationRepoId: githubInstallationRepoId,
  deployBranch,
});
console.log("Connected Git repository:", connection.id);

const updated = await client.apps.git.update(appId, {
  deployBranch,
  deploymentStatusEvents: true,
  pullRequestComments: true,
});
console.log("Updated Git connection:", updated.id);

// Uncomment to disconnect the repository from the app.
// await client.apps.git.del(appId);
